### PR TITLE
JSON Output Improvements

### PR DESCRIPTION
**Pull request by lawrencelomax**
- [x] Use Event Names in Event Sink
- [x] Add `UDID` to `FBSimulator` JSON.
- [x] Add runtime validation of objects that poport to be JSON serializable to prevent exception throwing
- [x] Fix busted JSON Serialization
- [x] Report Errors in JSON
- [x] Redirect `stderr` to `stdout` with logger events in JSON.
- [x] Replace `Writer` in a `Runner` with an `EventReporter`.
- [x] Remove `SuccessFailureWriter`
- [x] Simplify Event Reporting via Extensions on `EventReporter` bridging to `EventReporterSubject`
- [x] Create a `JSON` enum that conforms to `FBJSONDescribeable`, so that Swift-expressed values get the benefit of strong-typing.
- [x] Timestamp all the things.

### CODE REVIEW

File: FBSimulatorControl/Configuration/FBSimulatorControlGlobalConfiguration.m [hunk 1]
```diff
@@ -141,18 +132,43 @@ + (NSString *)description
...
 @end
 
 @implementation FBSimulatorControlGlobalConfiguration (Environment)
 
-+ (void)setStderrLoggingEnabled:(BOOL)enabled
++ (void)setDefaultLogger:(id<FBSimulatorLogger>)defaultLogger
 {
-  setenv(FBSimulatorControlStderrLogging.UTF8String, enabled ? "YES" : "NO", 1);
+  if (logger) {
+    [logger logFormat:@"The Default logger has allready been set to %@, %@ will have no effect", logger, NSStringFromSelector(_cmd)];
```
Comment (asm89) [inline FBSimulatorControl/Configuration/FBSimulatorControlGlobalConfiguration.m]: nit: allready

File: fbsimctl/FBSimulatorControlKit/Sources/Command.swift [hunk 2]
```diff
@@ -99,10 +98,21 @@ public func == (left: Command, right: Command) -> Bool {
...
+      switch (leftMaybeFormat, rightMaybeFormat) {
+      case (.Some(let leftFormat), .Some(let rightFormat)):
+        return leftFormat == rightFormat
+      case (.None, .None):
+        return true
+      default:
+        return false
+      }
+    case (.Create(let leftConfiguration, _), .Create(let rightConfiguration, _)):
+      return leftConfiguration == rightConfiguration//  && leftFormat == rightFormat
```
Comment (asm89) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: What's up with the commented out `&&` here?

Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/Command.swift]: Looks like I need to do the same workaround as in line 102 for nullable arrays.

File: FBSimulatorControl/Configuration/FBSimulatorControlGlobalConfiguration.m (hunk 1 above)
Comment (lawrencelomax) [inline FBSimulatorControl/Configuration/FBSimulatorControlGlobalConfiguration.m]: (y)

**Comment by lawrencelomax** [comment]
@facebook-github-bot shipit

**Pull request merged**